### PR TITLE
Add support for the "pull-request-branch-name.separator" config option

### PR DESCRIPTION
**Pull request by rhyskoedijk**
Changes:
- Add support for the "pull-request-branch-name.separator" config option.
- Update V1 -> V2 migration documentation with [list of changes to environment variables](https://github.com/tinglesoftware/dependabot-azure-devops/blob/c8b5fd6a8a8f6fd6b49acfa47171a64fe2c1a778/docs/migrations/v1-to-v2.md#changes-to-environment-variables).

Example using "-" as the branch name separator:
image

### CODE REVIEW

File: extension/tasks/dependabotV2/utils/dependabot-cli/DependabotOutputProcessor.ts [hunk 1]
```diff
@@ -95,16 +96,21 @@ export class DependabotOutputProcessor implements IDependabotUpdateOutputProcess
           return true;
         }
 
+        // TODO: Add config for custom PR description header text? Task V1 supported this via DEPENDABOT_MESSAGE_HEADER
+        // TODO: Add config for custom PR description footer text? Task V1 supported this via DEPENDABOT_MESSAGE_FOOTER
+        // TODO: Add config for GitHub compatibility score badges? Task V1 supported this via DEPENDABOT_COMPATIBILITY_SCORE_BADGE
+
```
Comment (mburumaxwell) [inline extension/tasks/dependabotV2/utils/dependabot-cli/DependabotOutputProcessor.ts:102]: I do not think we need to make these customizable at all.

**Review by mburumaxwell — CHANGES_REQUESTED** [review]
Let us remove support for custom header, footer, and compatibility score.

File: extension/tasks/dependabotV2/utils/dependabot-cli/DependabotOutputProcessor.ts (hunk 1 above)
Comment (rhyskoedijk) [reply extension/tasks/dependabotV2/utils/dependabot-cli/DependabotOutputProcessor.ts:102]:
```suggestion
```

File: docs/migrations/v1-to-v2.md [hunk 2]
```diff
@@ -42,13 +42,31 @@ This was a customisation/workaround specific to the V1 update script that can no
...
+The following environment variables have been removed entirely; the feature is no longer supported:
+
+| Removed Environment Variable | Reason |
+|--|--|
+|`DEPENDABOT_PR_NAME_PREFIX_STYLE`| Feature is not supported |
+|`DEPENDABOT_COMPATIBILITY_SCORE_BADGE`| Feature is not supported |
+|`DEPENDABOT_MESSAGE_HEADER`| Feature is not supported |
+|`DEPENDABOT_MESSAGE_FOOTER`| Feature is not supported |
+|`DEPENDABOT_SIGNATURE_KEY`| Feature is not supported |
+|`DEPENDABOT_JOB_ID`| Set automatically by extension |
```
Comment (rhyskoedijk) [inline docs/migrations/v1-to-v2.md:69]:
```suggestion
|`DEPENDABOT_PR_NAME_PREFIX_STYLE`| Feature is not supported; It is not an official configuration |
|`DEPENDABOT_COMPATIBILITY_SCORE_BADGE`| Feature is not supported; It is not an official configuration |
|`DEPENDABOT_MESSAGE_HEADER`| Feature is not supported; It is not an official configuration |
|`DEPENDABOT_MESSAGE_FOOTER`| Feature is not supported; It is not an official configuration |
|`DEPENDABOT_SIGNATURE_KEY`| Feature is not supported; It is not an official configuration |
|`DEPENDABOT_JOB_ID`| Set automatically by extension |
```

**Comment by rhyskoedijk** [comment]
> Let us remove support for custom header, footer, and compatibility score.

removed

**Review by mburumaxwell — APPROVED** [review]

**Pull request merged**